src/components/Filters.tsx

```diff
@@ -75,6 +77,16 @@ const Filters = ({
     .filter((v, i, a) => a.indexOf(v) === i)
     .sort();
 
+  const itemWorldOptions = items
+    .filter(byItemName(filters.itemName))
+    .filter(byUpgrade(filters.upgrade))
+    .filter(byRarity(filters.rarity))
+    .filter(byFavourite(filters.onlyFavourites))
+    .filter(byItemWorld(filters.itemWorld))
```

Comment:
you need to remove this line here, so when a world filter is set you are able to set it to a different value. but you should add this line to all other filter options (effectTargetOptions, upgradeOptions, rarityOptions). you should also add `.filter(byEffectTarget(filters.effectTarget))` so the itemWorldOptions gets narrowed down if a filter by effect target is set.